staging/src/k8s.io/kubectl/pkg/cmd/drain/drain.go

```diff
@@ -177,6 +179,40 @@ func (o *DrainCmdOptions) onPodDeletedOrEvicted(pod *corev1.Pod, usingEviction b
 	}
 }
 
+// onPodDeletionOrEvictionFailed is called by drain.Helper, when a pod is eviction/deletetion is failed
+func (o *DrainCmdOptions) onPodDeletionOrEvictionFailed(pod *corev1.Pod, usingEviction bool) {
```

Comment:
I don't think we want to always add that much information for all drains, I'd probably hide it behind at least `-v=2`, so add here:
```
if !klog.V(2).Enabled() {
    return
}
// the rest of method
```
Especially that you're printing the entire object, which is not desirable imo.